e2e/testsuite.yaml

```diff
@@ -84,3 +87,37 @@ items:
     method: DELETE
     header:
       Authorization: "Basic YWRtaW46MTIzNDU2"
+
+  # Notifications
+- name: createNotification
+  request:
+    api: /notification.halo.run/v1alpha1/notifications
+    method: POST
+    body: |
+      {
+        "spec": {
+          "recipient": "{{.param.userName}}",
+          "reason": "fake-reason",
+          "title": "test 评论了你的页面《关于我》",
+          "rawContent": "Fake raw content",
+          "htmlContent": "<p>Fake html content</p>",
+          "unread": true
+        },
+        "apiVersion": "notification.halo.run/v1alpha1",
+        "kind": "Notification",
+        "metadata": {
+          "name": "{{.param.notificationName}}"
+        }
+      }
+    header:
+      Content-Type: application/json
+      Authorization: "{{.param.auth}}"
+  expect:
+    statusCode: 201
+- name: deleteUserNotification
+  request:
+    api: |
+      /api.notification.halo.run/userspaces/{{.param.userName}}/v1alpha1/notifications/{{.param.notificationName}}
```

Comment:
Please make sure if the blew API is correct.

`http://halo:8090/apis/api.notification.halo.run/userspaces/ZjuBhd/v1alpha1/notifications/DYDgLi`

If it is correct, then I think you might need to query the generated name. Query it first, then get the name. See also https://github.com/halo-dev/halo/pull/4920/files#diff-f576994e2631a97d58544de7da11ec699440776f560913a49035a50bf6f5b361R180

Reply:
Done 😆
The tested API was written incorrectly